pkg/spoke/spokeagent.go

```diff
@@ -399,23 +433,39 @@ func (o *SpokeAgentOptions) getOrGenerateClusterAgentNames() (string, string) {
 		// and then load the cluster name from the mounted secret
 		clusterNameFilePath := path.Join(o.HubKubeconfigDir, hubclientcert.ClusterNameFile)
 		clusterNameBytes, err := ioutil.ReadFile(path.Clean(clusterNameFilePath))
-		if err != nil {
-			// generate random cluster name if faild
-			clusterName = generateClusterName()
-		} else {
+		switch {
+		case len(clusterNameInCert) > 0:
+			// use cluster name loaded from the tls certification
+			clusterName = clusterNameInCert
+			if err != nil && clusterNameInCert != string(clusterNameBytes) {
```

Comment:
this should be `err == nil`, and we may use a new var name for this error

Reply:
Fixed.